modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/Field/Field.es.js

```diff
@@ -147,25 +146,32 @@ export const Field = ({field, onBlur, onChange, onFocus, ...otherProps}) => {
 		<ErrorBoundary onError={() => setHasError(true)}>
 			<Suspense fallback={<ClayLoadingIndicator />}>
 				<ParentFieldContext.Provider value={field}>
-					<div
-						className="ddm-field"
-						data-field-name={field.fieldName}
-					>
-						<FieldLazy
-							visible
-							{...field}
-							{...otherProps}
-							onBlur={(event) =>
-								onBlur(mountStruct(event, field))
-							}
-							onChange={(event, value) =>
-								onChange(mountStruct(event, field, value))
-							}
-							onFocus={(event) =>
-								onFocus(mountStruct(event, field))
-							}
-						/>
-					</div>
+					<AutoFocus>
```

Comment:
If all fields are wrapped with `AutoFocus` how do we make sure only the first on the page will be focused? I'm sure you tested it, but I'm not sure I understand how it works.

Reply:
Yeah, the verification is quite simple in this case, it compares with the `document.activeElement` if the element is not inside the Form he considers looking for the element and focusing, and in the next subsequent renderings he ignores why there is a field focused.

We might not need to rely on `activeElement` by creating a Context just for that, but I thought it was a bit exaggerated.

Reply:
We are just relying on sequential rendering of React

Reply:
Ah ok. Now I get what 
```js
!containerElement.current.contains(document.activeElement)
````
is doing. The thing is that this is pretty slow to be done on every render for every field. Can't we just do that logic once when the page gets attached?

Reply:
I get it won't be done for every **render**. Not sure how often the dependencies `[childRef, containerElement, lazyStatus]` make that useLayoutEffect get called, but you get my point :)

Reply:
I think we can move this to PageRenderer, but we would not be able to track when the fields were actually rendered due to React.lazy.

Reply:
> I get that it won't be done for every render. Not sure how often the dependencies [childRef, containerElement, lazyStatus] make that useLayoutEffect get called, but you get my point :)

I ended up missing that message! Hmm yeah, that's a good point. I can't see another way to do this, I thought about checking the structure to know which is the first field but I think we will be exchanging one for the other.

Maybe just compare `document.activeElement` to `firstInput` in `containerElement`? This would relieve the use of `contains`.